ako-infra/avirest/handle_netinfo.go

```diff
@@ -0,0 +1,261 @@
+		if *lslrList[i].SegmentID == ls {
+			lslrList = append(lslrList[:i], lslrList[i+1:]...)
+			return true, lslrList
+		}
+	}
+	return false, lslrList
+}
+
+func matchSegmentInCloud(lslrList []*models.Tier1LogicalRouterInfo, lslrMap map[string]string) bool {
+	if len(lslrMap) != len(lslrList) {
+		return false
+	}
+	for i := range lslrList {
+		if lslrMap[*lslrList[i].SegmentID] != *lslrList[i].Tier1LrID {
+			return false
+		}
+	}
+	return true
+}
+
+func constructLsLrInCloud(lslrMap map[string]string) []*models.Tier1LogicalRouterInfo {
+	var lslrList []*models.Tier1LogicalRouterInfo
+	for ls, lr := range lslrMap {
+		lslr := models.Tier1LogicalRouterInfo{
+			SegmentID: &ls,
+			Tier1LrID: &lr,
+		}
+		lslrList = append(lslrList, &lslr)
+	}
+	return lslrList
+}
+
+func getAviCloudFromCache(client *clients.AviClient, cloudName string) (bool, models.Cloud) {
+	if CloudCache == nil {
+		if AviCloudCachePopulate(client, cloudName) != nil {
+			return false, models.Cloud{}
+		}
+	}
+	return true, *CloudCache
+}
+
+// AviCloudCachePopulate queries avi rest api to get cloud data and stores in CloudCache
+func AviCloudCachePopulate(client *clients.AviClient, cloudName string) error {
+	uri := "/api/cloud/?name=" + cloudName
+	result, err := lib.AviGetCollectionRaw(client, uri)
+	if err != nil {
+		utils.AviLog.Warnf("Cloud Get uri %v returned err %v", uri, err)
+		return err
+	}
+
+	elems := make([]json.RawMessage, result.Count)
+	err = json.Unmarshal(result.Results, &elems)
+	if err != nil {
+		utils.AviLog.Warnf("Failed to unmarshal data, err: %v", err)
+		return err
+	}
+
+	if result.Count != 1 {
+		utils.AviLog.Errorf("Expected one cloud for cloud name: %s, found: %d", cloudName, result.Count)
+		return fmt.Errorf("Expected one cloud for cloud name: %s, found: %d", cloudName, result.Count)
+	}
+
+	CloudCache = &models.Cloud{}
+	if err = json.Unmarshal(elems[0], &CloudCache); err != nil {
+		utils.AviLog.Warnf("Failed to unmarshal cloud data, err: %v", err)
+		return err
+	}
+	return nil
+}
+
+func executeRestOp(key string, client *clients.AviClient, restOp *utils.RestOp, retryNum ...int) {
+	utils.AviLog.Debugf("key: %s, Executing rest operation to sync object in cloud: %v", key, *restOp)
+	retry := 0
+	if len(retryNum) > 0 {
+		utils.AviLog.Infof("key: %s, Retrying to execute rest request", key)
+		retry = retryNum[0]
+		if retry >= 3 {
+			utils.AviLog.Errorf("key %s, rest request execution aborted after reqtrying 3 times", key)
+			return
+		}
+	}
+	err := rest.AviRestOperateWrapper(client, []*utils.RestOp{restOp})
+	if err != nil {
+		utils.AviLog.Infof("key %s, Got error in executing rest request: %v", key, err)
+		if checkAndRetry(key, err) {
+			executeRestOp(key, client, restOp, retry+1)
+		}
+	}
+	AviCloudCachePopulate(client, utils.CloudName)
+	utils.AviLog.Infof("key: %s, Successfully executed rest operation to sync object in cloud: %v", key, *restOp)
+}
+
+func checkAndRetry(key string, err error) bool {
+	if err == nil {
```

Comment:
checkAndRetry will be call only when err != nil. so i feel this "if" is redundant.

Reply:
This was a safety check, but we don't it for now, removed.